strax/storage/mongo.py

```diff
@@ -73,6 +73,39 @@ def get_metadata(self, key):
             return doc['metadata']
         raise strax.DataNotAvailable
 
+    def _build_chunk_registry(self, backend_key):
+        """
+        :param backend_key: strax.DataKey to query the collection for
+        Build chunk info in a single registry using only one query to
+        the database. This is much faster as one does not have to do
+        n-chunk queries to the database. Just one will do. As the
+        documents-size is limited to 16 MB, it's unlikely that we will
+        run into memory issues (that we otherwise would not run into).
+        """
+
+        query = backend_key_to_query(backend_key)
+        chunks_registry = self.db[self.col_name].find(
+            {**query, **{"chunk_i": {'$exists':True}}},
+            {"chunk_i": 1, "data": 1})
+        if chunks_registry is not None:
+            # We are going to convert this to a dictionary as that is
+            # easier to lookup
+            self.chunks_registry = {}
+        else:
+            raise strax.DataNotAvailable(f'Unable find data for {query}')
+
+        for doc in chunks_registry:
+            chunk_key = doc.get('chunk_i', None)
+            if chunk_key is None:
+                # Should not happen because of the projection in find
+                # but let's double check
+                raise ValueError(
+                    f'Projection failed, got doc with no "chunk_i":\n{doc}')
+            # Update our registry with this chunks info. Use chunk_i as
+            # chunk_key
+            self.chunks_registry[chunk_key] = doc.copy()
```

Comment:
Integers as dictionary keys can cause issues later if you try to json-encode it. Probably not an issue here, but something to remember.

Reply:
I changed it just to be on the safe side